zk/stages/stage_interhashes.go

```diff
@@ -130,7 +130,7 @@ func SpawnZkIntermediateHashesStage(s *stagedsync.StageState, u stagedsync.Unwin
 
 	eridb.OpenBatch(quit)
 
-	if s.BlockNumber == 0 || shouldRegenerate {
+	if (s.BlockNumber == 0 || shouldRegenerate) && !cfg.zk.IncrementTreeAlways {
```

Comment:
Would it make sense to have a check for the increment tree always just on its own, then the normal flow in an else for that?